jeaye/stdman
# Regenerate with v20220201 (from unofficial releases)

# v20220201

Man pages are regenerated with the HTML books from the unofficial source [1] (mentioned by the cppreferece.com in their website[2]) and not the official one.

## Release Notes
  >  v20220201
  >  - Update content from the website.

## Build Process

Here's is the details of build process, for those who may want to contribute:

- Note that, this build has been done on a Macbook Air M1. the prerequisites for this build are:
  - findutils
  - gsed
  - elinks

1. fork, clone and create a branch
```sh
git clone git@github.com/<username>/stdman && cd $_
git switch -C update/v20220201
```

2. download the latest html-book and extract it
```sh
wget -c https://github.com/PeterFeicht/cppreference-doc/releases/download/v20220201/html-book-20220201.zip
unzip html-book-20220201.zip -d html-book
```

3. create a directory for reference html files and mv html files to thre
```sh
mkdir -p reference/en/
mv html-books/reference/en/cpp/ reference/en/
```

4. configure and generate man pages
```sh
./configure --reference=./reference/en/cpp
make generate
```

5. install and test man pages (check the date at the end of a man page)
```sh
sudo make install
sudo /usr/libexec/locate.updatedb (macos)
man std::wstringstream
```

6. remove trailing whitespace from generated man pages
```sh
gfind man | xargs -I{} gsed -i'' 's/[[:blank:]]*$//' {} # macos
find man | xargs -I{} sed -i'' 's/[[:blank:]]*$//' {} # gnu
```

7. stage changes, clean untracked files and push changes to remote
```
git add -u && git clean -xfd
git commit && git push origin update/v20220201
```

[1]: https://github.com/PeterFeicht/cppreference-doc
[2]: https://en.cppreference.com/w/Cppreference:Archives
